compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/NodeParser.java

```diff
@@ -21,21 +21,32 @@
 import io.ballerina.compiler.internal.parser.ParserFactory;
 
 /**
- * Parses a given input and produces a {@code Node}  / {@code NodeList}.
+ * Parses a given input and produces a {@code Node} / {@code NodeList}.
  *
  * @since 1.3.0
  */
 public class NodeParser {
 
     /**
-     * Parses the input as statements.
+     * Parses the input as an action or expression.
      *
      * @param text the input
-     * @return a {@code NodeList<StatementNode>}
+     * @return an {@code ExpressionNode}
      */
-    public static NodeList<StatementNode> parseStatements(String text) {
+    public static ExpressionNode parseActionOrExpression(String text) {
         BallerinaParser parser = ParserFactory.getParser(text);
-        return new NodeList<>(parser.parseAsStatements().createUnlinkedFacade());
+        return parser.parseAsActionOrExpression().createUnlinkedFacade();
+    }
+
+    /**
+     * Parses the input as a binding pattern.
+     *
+     * @param text the input
+     * @return a {@code BindingPatternNode}
+     */
+    public static BindingPatternNode parseBindingPattern(String text) {
+        BallerinaParser parser = ParserFactory.getParser(text);
```

Comment:
Can we keep one global instance of `BallerinaParser` for `NodeParser` class? We need to empty the context stack after each parse if we are doing so

Reply:
Resolving as we discussed offline.